Scraper2/Scrapper2.cpp

```diff
@@ -0,0 +1,184 @@
+#include <iostream>
+#include <SFML/Window.hpp>
+#include <SFML/Graphics.hpp>
+#include <algorithm>
+#include <thread>
+
+#include "WindowHelper.h"
+#include "Player.h"
+#include "Enemy.h"
+#include "ProjectileType.h"
+#include "CollisionHandler.h"
+#include "GameMenu.h"
+
+CollisionHandler collisionHandler;
+
+void handleSystemEvents(sf::RenderWindow& window, bool& isGameRunning, bool& isGamePaused, GameMenu& menu)
+{
+    sf::Event event;
+    while (window.pollEvent(event))
+    {
+        switch (event.type)
+        {
+        case sf::Event::Closed:
+            window.close();
+            break;
+        case sf::Event::KeyPressed:
+            if (event.key.code == sf::Keyboard::Escape)
+            {
+                isGamePaused = !isGamePaused;
+                if (isGamePaused) 
+                {
+                    menu.draw(window);
```

Comment:
It would be best if you didn't draw right here. It makes it hard to track what part of the Game Tick is executed in which function and in which sequence
see Game Loop pattern: https://gameprogrammingpatterns.com/game-loop.html